src/main/java/com/google/sps/servlets/SearchMealServlet.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+
+@WebServlet("/search-meal")
+public class SearchMealServlet extends HttpServlet {
+
+    private ArrayList<String> dishes;
+    private ArrayList<String> result;
+
+    @Override
+    public void init() {
+        dishes = new ArrayList<>();
+        dishes.add("fried potato with mushrooms and onion");
+        dishes.add("pizza with pineaple, sausage and tomato");
+        dishes.add("vegetable soup with onion");
+        dishes.add("chocolate cake with butter cream and strawberry");
+        dishes.add("hot chocolate with sugar and caramel");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String gson = new Gson().toJson(result);
+
+        response.setContentType("application/json");
+        response.getWriter().println(gson);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        result = searchDishByKeyWord(getParameter(request, "text", ""));
```

Comment:
Note, that this will probably return the whole list of dishes. In the long term we'd probably want to limit the results (kinda like the Google search engine - oftentimes you get millions of matches, but you receive only the top 20; the response would've been far too big otherwise).